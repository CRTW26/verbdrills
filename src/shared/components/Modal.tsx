import React from 'react'
import { VscChromeClose } from 'react-icons/vsc'

interface Props {
  children: JSX.Element
  onClose: () => void
}

const Modal: React.FC<Props> = ({ children, onClose }) => {
  return (
    <div className="modal">
      <div className="modal__inner">
        <button onClick={onClose} className="modal-close">
          <VscChromeClose color="#e0ff4f" fontSize={'2rem'} />
        </button>
        {children}
      </div>

      <style jsx>{`
        .modal-close {
          position: absolute;
          top: 4px;
          right: 4px;
          background: none;
          border: none;
        }

        button:hover {
          cursor: pointer;
        }

        .modal {
          position: fixed;
          display: flex;
          justify-content: center;
          align-items: start;
          top: 0;
          left: 0;
          bottom: 0;
          right: 0;
          z-index: 999;
        }

        .modal::before {
          content: '';
          position: absolute;
          width: 100%;
          height: 100%;
          background: black;
          opacity: 0.5;
        }

        .modal__inner {
          position: absolute;
          background: #0b3954;
          opacity: 1;
          width: 100%;
          padding: 1rem;
        }

        @media screen and (min-width: 720px) {
          .modal__inner {
            margin-top: 8rem;
            max-width: 580px;
            border-radius: 5px;
          }
        }
      `}</style>
    </div>
  )
}

export default Modal
