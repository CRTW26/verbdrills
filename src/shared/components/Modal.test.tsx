import { render } from '@testing-library/react'
import Modal from './Modal'

describe('<Modal />', () => {
  test('should render', () => {
    const { asFragment } = render(
      <Modal onClose={jest.fn()}>
        <h1>Test</h1>
      </Modal>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
