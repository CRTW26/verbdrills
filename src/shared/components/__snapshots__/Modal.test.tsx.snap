// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render 1`] = `
<DocumentFragment>
  <div
    class="jsx-1277463556 modal"
  >
    <div
      class="jsx-1277463556 modal__inner"
    >
      <button
        class="jsx-1277463556 modal-close"
      >
        <svg
          color="#e0ff4f"
          fill="currentColor"
          font-size="2rem"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(224, 255, 79);"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M7.116 8l-4.558 4.558.884.884L8 8.884l4.558 4.558.884-.884L8.884 8l4.558-4.558-.884-.884L8 7.116 3.442 2.558l-.884.884L7.116 8z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
      <h1>
        Test
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
